cmd/kube-proxy/app/server.go

```diff
@@ -631,6 +632,17 @@ func newProxyServer(logger klog.Logger, config *kubeproxyconfig.KubeProxyConfigu
 	rawNodeIPs := getNodeIPs(logger, s.Client, s.Hostname)
 	s.PrimaryIPFamily, s.NodeIPs = detectNodeIPs(logger, rawNodeIPs, config.BindAddress)
 
+	if config.NodePortAddressesPrimary {
+		var nodePortAddresses []string
+		if nodeIP := s.NodeIPs[v1.IPv4Protocol]; nodeIP != nil {
+			nodePortAddresses = append(nodePortAddresses, fmt.Sprintf("%s/32", nodeIP.String()))
+		}
+		if nodeIP := s.NodeIPs[v1.IPv6Protocol]; nodeIP != nil {
+			nodePortAddresses = append(nodePortAddresses, fmt.Sprintf("%s/128", nodeIP.String()))
+		}
```

Comment:
I noticed kube-proxy now keeps logging the following error every sync in a regular IPv4 cluster, which should be introduced by https://github.com/kubernetes/kubernetes/pull/122724. And the error persists with this PR.
```
E0205 04:03:28.742266       1 proxier.go:508] "--nodeport-addresses includes localhost but localhost NodePorts are not supported" address="::1"
E0205 04:03:58.625792       1 proxier.go:508] "--nodeport-addresses includes localhost but localhost NodePorts are not supported" address="::1"
E0205 04:04:28.666855       1 proxier.go:508] "--nodeport-addresses includes localhost but localhost NodePorts are not supported" address="::1"
```
Since the new option `NodePortAddressesPrimary` makes it more clear that only primary IP will be accepted, should it stop adding the IPv6 loopback address to nodePortAddresses in this case? However, when nodePortAddresses is used, no CIDRs of the given family means the CIDR "0.0.0.0/0" or "::/0" will be added, so it seems more changes are needed.

Reply:
oh, oops, we definitely shouldn't be adding `::1/128` to nodePortAddresses...